services/src/main/java/org/keycloak/userprofile/validator/DuplicateUsernameValidator.java

```diff
@@ -64,12 +64,21 @@ public ValidationContext validate(Object input, String inputHint, ValidationCont
         KeycloakSession session = context.getSession();
         UserModel existing = session.users().getUserByUsername(session.getContext().getRealm(), value);
         UserModel user = UserProfileAttributeValidationContext.from(context).getAttributeContext().getUser();
+        String valueLowercased = value.toLowerCase();
 
-        if (! KeycloakModelUtils.isUsernameCaseSensitive(session.getContext().getRealm())) value = value.toLowerCase();
+        if (! KeycloakModelUtils.isUsernameCaseSensitive(session.getContext().getRealm())) value = valueLowercased;
```

Comment:
This is different than what we are doing here https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/userprofile/validator/UsernameMutationValidator.java#L69.

If I read that code correctly, only if the realm is not set to use case-sensitive usernames we should lower case the value.

This was added by #14678. @vramik Wdyt?

Reply:
@pedroigor I think I'm doing the same than before, but just saving the lower-cased value for the email comparison (email is always lower-cased, so we should search for email with the lowercased username).

Reply:
You're right. Sorry for the noise.